### PR TITLE
chore(semantic): copy tests from typescript-eslint’s scope-manager

### LINKED ISSUE #2947: Robust testing for scope analysis
This issue is follow-up of https://github.com/oxc-project/oxc/issues/1227 and https://github.com/oxc-project/oxc/pull/2498.

## Status quo

oxc currently rely on test262 and manual testing.

test262:
https://github.com/oxc-project/oxc/blob/1519b9000bcdc7c43c79c8b26e194a13f7df1f1e/crates/oxc_semantic/tests/scopes.rs

manual testing:
https://github.com/oxc-project/oxc/blob/e2d2ce3369bd49743cf95c095c206a92dfe0acda/tasks/coverage/src/test262.rs#L212-L235

- test262 only ensure the all indentifier are resolved, doesn't ensure correctness.
- Manual testing ensure correctness but only have a few cases.

## Problems

With the above summary, the current scope analysis is running in half dark.
- Can't not detect effects/changes. If someone changes scope analysis code, we could only rely on the few manual tests to ensure the correctness.
- Dx of writing Manual tests is bad. The API could handle simple cases well. But we have to admit that, as much as we want to try make the test case simple, there's a bigger chance we need to handle complicated code. Writing manual tests for them seems impossible with current test API.

## Solutions

1. We assume the current scope analysis behaviors are correct and visualize them in snapshot testing.
2. Visualize scope information of tests262 and also manual tests.

If someone change scope analysis code, we could review if it is correct by review changes of snapshot.

### PR DESCRIPTION
**Pull request by Dunqing**
close:  #2947

These fixtures were copied from https://github.com/typescript-eslint/typescript-eslint/tree/a5b652da1ebb09ecbca8f4b032bf05ebc0e03dc9/packages/scope-manager/tests/fixtures. We used them to test out semantic `ScopeTree` and `SymbolTable`

### CODE REVIEW

**Comment by Dunqing** [comment]
* **#3990** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/3990?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Dunqing and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by Dunqing** [comment]
It's a bit strange that my local and ci's snap has the fields in a different order. I ended up adjusting the `push_str` order of the different fields to get ci to pass

**Comment by overlookmotel** [comment]
This would be a pain in the short term, but maybe save us time overall by allowing us to find bugs faster:

Could we parse the snapshots from `typescript-eslint` repo ([example](https://github.com/typescript-eslint/typescript-eslint/blob/a5b652da1ebb09ecbca8f4b032bf05ebc0e03dc9/packages/scope-manager/tests/fixtures/block/inherited-scope.ts.shot)) and check them against our own scopes tree? The format is not the same, but it's *mostly* the same info.

**Comment by Dunqing** [comment]
> 
> Could we parse the snapshots from `typescript-eslint` repo ([example](https://github.com/typescript-eslint/typescript-eslint/blob/a5b652da1ebb09ecbca8f4b032bf05ebc0e03dc9/packages/scope-manager/tests/fixtures/block/inherited-scope.ts.shot)) and check them against our own scopes tree? The format is not the same, but it's _mostly_ the same info.

I gave it a try, it’s hard to parse to `JSON`
